### PR TITLE
Update dependencies, add code coverage, update linting

### PR DESCRIPTION
**Pull request by HughePaul**
Update to latest dependencies,
Add Snyk vulnerability check,
Add linting rules,
Add editor config,
Add code coverage and rules, but don't enforce as a CI test yet,
Add more tests, fix some linting in the code
Test on node 4 and 6

### CODE REVIEW

File: test/spec/spec.form.js [hunk 1]
```diff
@@ -292,6 +292,30 @@ describe('Form Controller', function () {
             });
         });
 
+        it('return an error if an unknown validator is specified', function () {
```
Comment (gavboulton) [inline test/spec/spec.form.js]: 'returns'.

File: test/spec/spec.form.js [hunk 2]
```diff
@@ -292,6 +292,30 @@ describe('Form Controller', function () {
...
+                template: 'index',
+                fields: {
+                    field: { validate: 'unknown' }
+                }
+            });
+            form.post(req, res, cb);
+            cb.should.have.been.calledWithExactly(new Error('Undefined validator:unknown'));
+        });
+
+        it('ignore an unkown formatter', function () {
```
Comment (gavboulton) [inline test/spec/spec.form.js]: 'ignores'.

**Comment by gavboulton** [comment]
This looks good. There's just a couple of minor comments on the tests.

**Comment by HughePaul** [comment]
Updated spelling on tests

**Comment by gavboulton** [comment]
Cool. @easternbloc @JoeChapman @joefitter @daniel-ac-martin?

File: test/spec/spec.validators.js [hunk 3]
```diff
@@ -271,7 +271,8 @@ describe('Validators', function () {
                 '07812/123/456',
                 '(07812)123456',
                 'mymobile',
-                078121223456
+                '078121223456',
+                78121223456
```
Comment (easternbloc) [inline test/spec/spec.validators.js]: not sure how valid this is but meh

**Comment by easternbloc** [comment]
LGTM

**Comment by easternbloc** [comment]
happy for you to merge @gavboulton

File: .eslintrc [hunk 4]
```diff
@@ -2,12 +2,20 @@
...
+        "semi": [2, "always"],
+        "brace-style": [2, "1tbs", { "allowSingleLine": true }],
+        "keyword-spacing": 2,
+        "space-before-blocks": 2,
+        "space-before-function-paren": [2, { "anonymous": "always", "named": "never" }],
+        "no-mixed-spaces-and-tabs": 2,
+        "comma-spacing": [2, {"before": false, "after": true}],
+        "key-spacing": [2, {"beforeColon": false, "afterColon": true}]
     }
 }
```
Comment (JoeChapman) [inline .eslintrc]: Thought about using eslint-config-homeoffice?
```
extends:
- "homeoffice/config/default"
env:
  es6: true
```
certainly cuts down a lot of "stuff"

Comment (HughePaul) [inline .eslintrc]: That only supports eslint 0.23.0. We could make a 2.x branch and translate all the rules.

File: package.json [hunk 5]
```diff
@@ -4,9 +4,12 @@
   "description": "",
   "main": "index.js",
   "scripts": {
-    "test": "npm run lint && npm run unit",
+    "test": "npm run lint && npm run unit && npm run cover && npm run snyk",
     "unit": "mocha --recursive test/spec/ --require test/helper --timeout 300000",
-    "lint": "eslint ./lib/ ./index.js"
+    "cover": "istanbul cover _mocha -- --recursive test/spec/ --require test/helper --timeout 300000",
+    "check-coverage": "istanbul check-coverage --statement 100 --branch 100 --function 100 --line 100",
```
Comment (daniel-ac-martin) [inline package.json]: I'm guessing we don't actually have this much coverage which is why you don't have check-coverage in the script for test.

If that's the case what do people think about setting the limits to just below what we currently have, that way we can at least ensure that we don't go backwards and can slowly 'ratchet' the limits up over time?

Comment (HughePaul) [inline package.json]: This is a good point, I'll update that.

**Comment by daniel-ac-martin** [comment]
LGTM.

File: .eslintrc (hunk 4 above)
Comment (gavboulton) [inline .eslintrc]: Are you happy with this @JoeChapman?

Comment (JoeChapman) [inline .eslintrc]: it's ok, we need to update that module though

**Pull request merged**